modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -464,9 +465,10 @@ public static ExtensionModel loadExtension(Class<?> clazz, Map<String, Object> p
     return new DefaultJavaExtensionModelLoader().loadExtensionModel(clazz.getClassLoader(), dslResolvingContext, params);
   }
 
-  public static String getImplicitConfigurationProviderName(ExtensionModel extensionModel,
+  public static String getImplicitConfigurationProviderName(String prefix, ExtensionModel extensionModel,
                                                             ConfigurationModel implicitConfigurationModel) {
-    return format("%s-%s-implicit", extensionModel.getName(), implicitConfigurationModel.getName());
+    Preconditions.checkArgument(isNotEmpty(prefix), "A prefix must be provided");
+    return format("%s-%s-%s-implicit", prefix, extensionModel.getName(), implicitConfigurationModel.getName());
```

Comment:
doesn't this break for the domain use case?

Reply:
Yes, it was breaking the domain use case. Code has been refactored in order to avoid this.